### ISSUE TITLE
[data-generation] Separate unbiased evaluation populations from stratified and policy-induced training acquisition

### ISSUE THREAD

**Issue by JanDuchscherer104** [JanDuchscherer104/ARIA-NBV#81]
Parent: #67

Depends on: #79–#80 for a validated oracle; #68 for root/replica identity; #73 for support/difficulty strata.

## Outcome

Define and implement a principled **task/state acquisition contract** for generating useful training data without corrupting the held-out estimand.

The pipeline should maintain two visibly separate distributions:

1. a frozen population/evaluation sample that estimates performance for the declared scene/target/root population;
2. an enriched training acquisition stream that deliberately oversamples rare, difficult, ambiguous, policy-induced, or underrepresented states.

Every generated row must retain how and why it entered the corpus. Confirmatory evaluation must never be filtered using future RRI, lookahead headroom, learned-policy error, or any other outcome unavailable at task admission.

## Why this issue is needed

Adding roots, replicas, candidate families, oracle lookahead, and online policies can produce a very large but highly nonuniform corpus. Without an explicit sampling design:

- dense scenes, large objects, or easy target crops can dominate row counts;
- rare target scales, distances, occlusions, and low-support roots remain underrepresented;
- policy rollouts increasingly visit states unlike the original logged-root distribution;
- hard-example mining can create selection bias and destroy population interpretability;
- a large number of correlated roots/candidates may be mistaken for independent evidence;
- training and evaluation distributions can become entangled through data-dependent filtering.

The thesis already requires scene-disjoint splits, denominators for exclusion strata, and scene-level primary aggregation. This issue deepens that contract for data generation.

### Active learning with sampling-bias correction

Importance Weighted Active Learning explicitly addresses the sampling bias introduced by selective label acquisition through importance weighting:

- Beygelzimer, Dasgupta & Langford, *Importance Weighted Active Learning*: https://arxiv.org/abs/0812.4952

### Core-set selection

Sener & Savarese formulate active selection as finding a representative core-set, motivating coverage of a state/feature space rather than repeatedly sampling dense modes:

- *Active Learning for Convolutional Neural Networks: A Core-Set Approach*, ICLR 2018: https://openreview.net/forum?id=H1aIuk-RW

### Dataset cartography

Swayamdipta et al. partition examples by training dynamics into easy, ambiguous, and hard regions. Their results motivate retaining ambiguity and inspecting persistently hard examples for label/pathology rather than treating one scalar difficulty score as truth:

- *Dataset Cartography: Mapping and Diagnosing Datasets with Training Dynamics*: https://aclanthology.org/2020.emnlp-main.746/

### Sequential state-distribution shift

DAgger is relevant once learned policies generate successor states: sequential prediction errors induce states outside the expert/offline distribution, so data aggregation from learner-induced trajectories is needed:

- Ross, Gordon & Bagnell, *A Reduction of Imitation Learning and Structured Prediction to No-Regret Online Learning*: https://proceedings.mlr.press/v15/ross11a.html

These sources motivate the design; they do not prescribe ARIA-NBV's strata or scientific estimand.

### A. Frozen evaluation population

Freeze before model/profile inspection:

```text
scene split and scene weights
source/snippet eligibility
target-task definition and cap
root-variant distribution
candidate-replica distribution
candidate/oracle contracts
failure and exclusion handling
acquisition budget
```

Evaluation sampling must use only admission-time information. Near-solved, low-support, negative-gain, and oracle-failure strata remain visible with denominators; future reward/headroom cannot decide whether a task belongs to the population.

### B. Enriched training acquisition

Training may intentionally alter sampling based on available strata, uncertainty, model error, or learner visitation. Such rows must carry an acquisition-policy identity and must not be analyzed as an unbiased sample of the evaluation population.

## Required strata

At minimum define typed, versioned bins for admission-time facts:

```text
scene
semantic class
target metric size and aspect ratio
target distance and relative bearing
logged root stage / trajectory quantile
actor-visible target confidence/support
root target support and root error stratum (oracle-only training stratum where declared)
local scene clutter/clearance
candidate valid count and family survival
occlusion/visibility difficulty (oracle diagnostic only unless actor-derived)
near-solved/saturation stratum
oracle-label availability/failure reason
```

Do not use arbitrary quantiles recomputed independently for every run. Fit bin edges on the training population, version them, and apply frozen edges to validation/test.

## Acquisition modes

Add a typed enum and persisted contract:

```python
class TaskAcquisitionMode(StrEnum):
    POPULATION_RANDOM = "population_random"
    STRATIFIED_BALANCED = "stratified_balanced"
    CORE_SET = "core_set"
    MODEL_UNCERTAINTY = "model_uncertainty"
    MODEL_ERROR = "model_error"
    POLICY_INDUCED = "policy_induced"
    DAGGER_AGGREGATION = "dagger_aggregation"
    MANUAL_AUDIT = "manual_audit"
```

Every admitted source/target/root/replica records:

```text
population key
acquisition mode/version
stratum IDs
selection score and availability
selection probability or deterministic rule
inclusion probability when known
acquisition checkpoint/model hash where applicable
acquisition round
parent episode/policy identity
```

## Recommended initial training mixture

Start with a transparent mixture rather than one fully adaptive stream:

```text
50% population-random rows
25% inverse-frequency stratified rows
15% support/difficulty core-set rows
10% learner-error or policy-induced rows
```

These proportions are pilot knobs, not canonical values. Keep a nonzero population-random anchor throughout training so performance does not optimize only an actively selected slice.

## Core-set representation

Use a compact acquisition embedding built from typed, non-label-leaking descriptors such as:

```text
target class/size/distance/bearing
root stage and history length
actor support summaries
candidate-family survival/diversity summaries
scene geometry/support summaries
```

For actor-facing training selection, exclude oracle RRI and GT target visibility. A separate oracle-diagnostic core-set may be built for metric validation but must be named privileged.

Prefer a deterministic k-center/farthest-first baseline before learned acquisition. It is easy to audit and aligns with the core-set objective.

## Model-driven acquisition

When a scorer exists, distinguish:

### Uncertainty sampling

Use predictive uncertainty/calibration over valid candidates or selected policy values. Uncertainty is not difficulty and can be miscalibrated; always retain diversity/stratum constraints.

### Error sampling

Use held-out or cross-fitted residuals, ranking regret, or endpoint error. Never use training loss from a model that trained on the same example as an unbiased difficulty label.

### Training-dynamics cartography

Persist per-example statistics across checkpoints:

```text
mean confidence/value error
variance across epochs
selection/ranking consistency
feasibility prediction consistency
```

Use these to identify:

- easy/redundant examples suitable for downsampling;
- ambiguous examples useful for decision-boundary learning;
- persistently hard examples requiring label/provenance inspection.

Do not automatically discard hard examples; they may represent rare geometry or broken labels.

## Policy-induced and DAgger-style data

Once #74–#75 are available:

1. retain the frozen offline/population corpus;
2. run the current learner on training scenes only;
3. collect learner-induced states and exact behavior provenance;
4. query the hard oracle densely or selectively;
5. add those states to a new acquisition round;
6. retrain from a mixture of old and new data;
7. never add validation/test policy trajectories to training.

Each successor state records its policy checkpoint, proposal contract, candidate table, and parent transition. DAgger-style collection expands training support; it does not alter the fixed confirmatory evaluation population.

## Importance weighting and estimands

Where acquisition probabilities are known and stable, expose optional inverse-probability weights:

```math
w_i = \frac{p_{population}(i)}{p_{train-acquisition}(i)}.
```

Use clipping/stabilization and report effective sample size. Do not invent probabilities for deterministic core-set or hard-mining selections; instead retain the full rule and evaluate on the untouched population holdout.

Primary policy evaluation remains unweighted scene-paired analysis on the frozen held-out population. Importance weighting is a training or secondary population-estimation tool, not a substitute for clean evaluation.

## Correlation and storage policy

- Roots, targets, candidate replicas, and rollout states from one scene remain in one split.
- Primary statistical unit remains the scene.
- Store stable grouping IDs: scene, source, target, root, replica, episode, chain, and acquisition round.
- Limit repeated near-identical states through state-space deduplication/core-set distance rather than row-count caps alone.
- Report both physical task count and raw candidate/transition count.

## Required artifacts

Create:

### `population-manifest`

Frozen eligibility, scene weights, target/root distributions, exclusions, and hashes.

### `training-acquisition-manifest`

Round-by-round selection rules, models/checkpoints, strata, probabilities, counts, and source population binding.

### `acquisition-audit`

Population versus acquired distributions, coverage gaps, effective sample size where applicable, duplicate/correlation summaries, and label-failure strata.

## Acceptance criteria

- [ ] Evaluation and training acquisition populations are separate typed artifacts.
- [ ] Confirmatory task admission never depends on future RRI, headroom, learned error, or policy outcome.
- [ ] Training supports population-random, stratified, core-set, and policy-induced acquisition modes.
- [ ] Every acquired row has exact mode, round, stratum, and selection provenance.
- [ ] Bin edges and stratum definitions are fit once on training and frozen for validation/test.
- [ ] A nonzero population-random training anchor is retained in adaptive rounds.
- [ ] Scene splits remain disjoint under roots, replicas, targets, and learner-induced trajectories.
- [ ] Raw row counts are never reported as independent sample size; scene/group identities are preserved.
- [ ] Model-driven acquisition uses cross-fitted or held-out signals where required.
- [ ] Importance weights are used only when their probability semantics are valid and are accompanied by effective-sample-size diagnostics.
- [ ] Validation/test data never enter DAgger or online training aggregation.

## Required tests

- future-label/outcome leakage at population admission;
- scene containment across acquisition rounds;
- frozen-bin application and unseen/out-of-range values;
- deterministic stratified/core-set selection;
- inclusion-probability and weight calculations on known fixtures;
- zero/near-zero acquisition probability safeguards;
- acquisition-manifest replay;
- policy-checkpoint and parent-transition provenance;
- duplicate/correlation grouping;
- train-only DAgger aggregation;
- untouched held-out evaluation manifest.

## Evidence before promotion

Report population versus training-acquired distributions for:

```text
scene and target class
target size/distance/stage
support and candidate validity
family survival and proposal regret
oracle failures and saturation
model uncertainty/error strata
policy-induced state distance
```

Also report:

```text
training efficiency per oracle query
held-out ranking/endpoint performance
rare-stratum performance
calibration
population-random versus enriched ablation
core-set/uncertainty/hard-mining overlap
label-error rate among persistently hard examples
```

## Relevant repository owners

- source/target/root campaign planning
- source and split manifests
- rollout episode and acquisition lineage
- candidate-support benchmark (#73)
- online environment/collector (#74)
- model training readers and samplers
- report-bundle population and exclusion tables

## Non-goals

- Do not claim that balanced training rows represent real-world prevalence.
- Do not remove hard or failed tasks merely to improve average metrics.
- Do not use validation/test trajectories for adaptive collection.
- Do not let row-level abundance override scene-level evaluation.
